src/lib/mojaloop-requests/mojaloopRequests.js

```diff
@@ -273,8 +272,12 @@ class MojaloopRequests {
         if(method.toUpperCase() !== 'PUT') {
             headers['accept'] = `application/vnd.interoperability.${resourceType}+json;version=1.0`;
         }
-
-        return headers;
+        if(span) {
+            const request = await span.injectContextToHttpRequest({headers});
```

Comment:
@mdebarros Not a big problem, and just my opinion but I'm not super keen on the pattern of allowing an unknown external to (potentially) modify the headers. My feeling is that it would be better for span to return header deltas for the enclosing function to merge. In this case, you are hiding what span may be doing with the headers.

Also looks like the injectContextToHttpRequest method on span is only dealing with headers (otherwise it would not work in this case, right?) so it seems wrong for it to be concerning itself with an HttpRequest object and not just a headers object.

Reply:
> @mdebarros Not a big problem, and just my opinion but I'm not super keen on the pattern of allowing an unknown external to (potentially) modify the headers. My feeling is that it would be better for span to return header deltas for the enclosing function to merge. In this case, you are hiding what span may be doing with the headers.
> 
> Also looks like the injectContextToHttpRequest method on span is only dealing with headers (otherwise it would not work in this case, right?) so it seems wrong for it to be concerning itself with an HttpRequest object and not just a headers object.

That is correct. Only trace headers (e.g. traceId & tracestate, etc) are being injected. I think the idea was that that the event-sdk would take most of the effort/complexity in doing the injection. I am happy to create a follow-up story to re-look at the approach for injecting the context.

Reply:
@mdebarros If this change is needed only to inject additional headers, then I think it would be better to pass those headers via object to `_buildHeaders`, for example:
```
async _buildHeaders (method, resourceType, dest, extraHeaders) {
...
return {
    ...headers,
    ...extraHeaders,
  }
}
```
Also, `span` could be renamed to `extraHeaders` in all other methods for better clarity.

Reply:
> @mdebarros If this change is needed only to inject additional headers, then I think it would be better to pass those headers via object to `_buildHeaders`, for example:
> 
> ```
> async _buildHeaders (method, resourceType, dest, extraHeaders) {
> ...
> return {
>     ...headers,
>     ...extraHeaders,
>   }
> }
> ```
> 
> Also, `span` could be renamed to `extraHeaders` in all other methods for better clarity.

Good suggestion. Will setup a story to re-factor the event-sdk to provide this capability, and will update this PR once that has been done.